otelcol/otelcoltest/config_test.go

```diff
@@ -70,6 +71,7 @@ func TestLoadConfigAndValidate(t *testing.T) {
 	factories, err := NopFactories()
 	assert.NoError(t, err)
 
+	require.NoError(t, featuregate.GlobalRegistry().Apply(map[string]bool{"otelcol.enableConnectors": true}))
```

Comment:
Should you `defer` to the previous state?

Reply:
Done.